Bug side avoidance

alrapal opened the pull request:
Solves and closes #45 
- There is now a method that returns a boolean for any given sensor (adapted from an already existing method for the front).
- After fixing, we could not reproduce the bug #45 anymore. Please try the code of this branch on your computer to see if it works on other machines, since it was supposed to be solved before.

alrapal committed:
Bug fix Side avoidance - #45
##########################################################
Type:
User story
Tags:
#45
Changes:
- Need to commit to try the changes on Chris' computer since the Mac version has problems when compiling
Notes:
- code done by myself and @gusaxtcha.

ChrisAxt committed:
Bug fix - side avoidance #45
##########################################################
Type:
Bug fix
Tags
BugFix, #45
Changes
- Changed frontIsClear() to IsClear(), adapted to check any given sensor on demand
- Side avoidance bug fix - stop turning when avoiding right.
Notes:
All changes done by myself and @alrapal. Committing for both due to Mac SMCE crashing.

iachella reviewed (changes requested):
- Line 314: Method description needs to be adapted to the changes (not depending on car speed/direction anymore, and also includes other sensors)
- Lines 331 to 334: Can be removed, if necessary we can get them back from older versions
- Line 337: Method description needs to be changed (not using isFrontClear() anymore)

Thank you!

alrapal commented:
Good catch, I make the changes and commit again =)

alrapal committed:
Bug fix Side avoidance - #45
##########################################################
Type:
User story
Tags:
#45
Changes:
- Changes requested by @iachella on PR #54:
 * Updated descriptions of methods
 * Removed unused code
Notes:

iachella reviewed (approved):
Looks good, nice methods description!
